research-software-directory/RSD-as-a-service
# News section

# News - admin section

**This is admin section of news. I still need to create view section on the home page.**

Closes #1059 

Changes proposed in this pull request:
* Added news edit section 
* News overview section with grid/list layout and add button for admins
* View/Edit menu button in the card and list item for admins 
* Included publishing date in the news url and as unique key (slug+publishing date)
* Increased size of image in the list item (shared component, it impacts all overviews)
* Fix invisible button border for the homepage buttons on firefox
* Added latest 3 news items to homepage incl. the button to news overview
* Added support for multiple images per news item
* Added news section in the documentation to administration chapter.

How to test:
* `make start` to build app
* login as rsd admin
* use + button at the top right and select "Add news" option (rsd-admin only)
* create new news item and upload few images
* try copy link and add image somewhere in the markdown
* publish news item, it will be shown in the news overview
* edit news article
* delete one of the images and confirm that link is removed from markdown
* delete news article
* validate that only published articles are shown on public page. Unpublish an article and logout. Confirm that article is not listed.
* in the user menu navigate to news section (rsd-admin role only)

## News menu option (for admins)

![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/4d71c62f-634b-4e67-abb0-e37edc1db713)

## Create news item
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/f4aa6d3f-cb9f-4671-9717-88bffb2cfab5)

## News items overview with add news button (for admins)
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/29b7f8b0-5703-4bf2-93dc-f4ff28e1470a)

## Edit news item
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/cb262de7-900c-4148-a039-9779c3e2c3a6)

## News article
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/787073cd-4187-4917-8417-55a750451124)

## Latest news on homepage
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/ebc6e16f-9f94-4b7d-a111-ecc2fbc4ca94)

PR Checklist:
